python/google/protobuf/service_reflection.py

```diff
@@ -133,7 +133,7 @@ def __init__(self, service_descriptor):
     """
     self.descriptor = service_descriptor
 
-  def BuildService(self, cls):
+  def BuildService(builder, cls):
```

Comment:
Not a Python expert, but isn't self a keyword to refer to the current object? That is, I don't think you can just change the name on this.

Reply:
`self` is actually a convention, not a keyword
```
>>> import keyword
>>> keyword.kwlist
['False', 'None', 'True', '__peg_parser__', 'and', 'as', 'assert', 'async', 'await', 'break', 'class', 'continue', 'def', 'del', 'elif', 'else', 'except', 'finally', 'for', 'from', 'global', 'if', 'import', 'in', 'is', 'lambda', 'nonlocal', 'not', 'or', 'pass', 'raise', 'return', 'try', 'while', 'with', 'yield']
```

In this specific case, I think it's actually clarifying, since inside the function we're adding methods to the object being built - and need to refer to those argument as `self`. Rather than aliasing the outer `self` of the builder, we're able to just name it `builder`.

Broadly, I don't think this would be the modern way of writing `service_reflection.py` - as sticking methods as properties is a bit convoluted (and there's no enforcement that it matches the abstract base class) - but it's probably safer to do any major changes to `service_reflection.py`. Using something like https://docs.python.org/3/library/abc.html would help enforce that the subclasses actually implement the interface.

Reply:
The tests in `python/google/protobuf/internal/service_reflection_test.py` are actually great for testing this code.
```
cd python
python3 setup.py test
```